bin/compile

```diff
@@ -34,22 +34,11 @@ metrics::setup
 
 if is_app_dir "${APP_BUILD_DIR}"; then
 	BUILD_DIR="/tmp/scala_buildpack_build_dir"
-	mv "${APP_BUILD_DIR}" "${BUILD_DIR}"
+	ln -s "${APP_BUILD_DIR}" "${BUILD_DIR}"
```

Comment:
(optional) I find these variable names confusing, given in most other buildpacks the BUILD_DIR really does mean the actual build dir.  For example the `rm "${BUILD_DIR}"` later looks really bad until you know it's only the symlink!

I wonder if the symlink path should be named something else?

Reply:
They are extremely confusing and will be changed :) I wanted to keep this PR small without any extra changes to make sure the change on its own doesn't break anything (caching mostly, hence the extra tests added _before_ this PR). For the Scala buildpack I try to be as "step-by-step" as possible. Unsure if I've gone too far.